File: Java/Maximum Subarray.java
// 1ms, 88.15%; 41.8MB, 5.16%
class Solution {
    public int maxSubArray(int[] nums) {
        int res = Integer.MIN_VALUE, cur = 0;
        for (int n : nums) {
            cur = n + (cur > 0 ? cur : 0);
            res = Integer.max(res, cur);
        }
        return res;
    }
}

File: Go/Maximum Subarray.go
// 4ms, 94.72%; 3.3MB, 57.14%
func maxSubArray(nums []int) int {
	res, cur := nums[0], nums[0]
	for i := 1; i < len(nums); i++ {
		if cur < 0 {
			cur = nums[i]
		} else {
			cur += nums[i]
		}
		if res < cur {
			res = cur
		}
	}
	return res
}
